### PR TITLE
Remove callback guard

### PR DESCRIPTION
**Pull request by EPashkin**
Part of https://github.com/gtk-rs/glib/pull/250

cc @GuillaumeGomez, @sdroege

### CODE REVIEW

File: check_init_asserts [hunk 1]
```diff
@@ -10,8 +10,7 @@ while (/^(\N*)\V*fn\s+(\w+)\s*(<[^(]+>)?\s*(\([^{;]+)\{\N*\n^(\N*)$/gms) {
 		$first_line =~ /^\s*(
 			assert_initialized_main_thread |
 			assert_not_initialized |
-			skip_assert_initialized |
-			callback_guard
+			skip_assert_initialized
```
Comment (EPashkin) [inline check_init_asserts]: unsafe functions just not checked, so this line not worked anyway

File: src/rt.rs [hunk 2]
```diff
@@ -47,15 +47,6 @@ macro_rules! assert_not_initialized {
     )
 }
 
-macro_rules! callback_guard {
-    () => (
-        let _guard = ::glib::CallbackGuard::new();
-        if cfg!(debug_assertions) {
-            assert_initialized_main_thread!();
-        }
```
Comment (EPashkin) [inline src/rt.rs]: I slightly worry about this check, maybe it really needed

**Comment by sdroege** [comment]
Yes but it should not be in the callback guard macro. I thought there was another macro for that?

**Comment by EPashkin** [comment]
This macro named `assert_initialized_main_thread` ;)
I just not sure that we need check that all callback called from main thread in gtk.

**Comment by EPashkin** [comment]
Ok, then I think that PR is ready

**Comment by GuillaumeGomez** [comment]
I just hope gtk doesn't have another bug. :-/

Thanks @EPashkin and @sdroege!

**Pull request merged**